tmail-backend/mailbox/team-mailboxes/src/main/scala/com/linagora/tmail/team/TeamMailboxRepository.scala

```diff
@@ -137,27 +141,45 @@ class TeamMailboxRepositoryImpl @Inject()(mailboxManager: MailboxManager,
       .flatMapIterable(mailboxMetaData => TeamMailbox.from(mailboxMetaData.getPath))
       .distinct()
 
-  override def addMember(teamMailbox: TeamMailbox, user: Username): Publisher[Void] = {
+  override def addMember(teamMailbox: TeamMailbox, teamMailboxMember: TeamMailboxMember): Publisher[Void] = {
     val session = createSession(teamMailbox)
-    val memberSession = mailboxManager.createSystemSession(user)
+    val memberSession = mailboxManager.createSystemSession(teamMailboxMember.username)
     SMono.fromPublisher(exists(teamMailbox))
       .filter(teamMailboxExist => teamMailboxExist)
       .switchIfEmpty(SMono.error(TeamMailboxNotFoundException(teamMailbox)))
       .flatMapIterable(_ => teamMailbox.defaultMailboxPaths)
-      .flatMap(mailboxPath => addRightForMember(mailboxPath, user, session)
+      .flatMap(mailboxPath => addRightForMember(mailboxPath, teamMailboxMember.username, session, teamMailboxMember.role)
         .`then`(subscribeForMember(mailboxPath, memberSession)))
       .`then`()
   }
 
-  private def addRightForMember(path: MailboxPath, user: Username, session: MailboxSession): SMono[Unit] =
+  private def addRightForMember(path: MailboxPath, user: Username, session: MailboxSession, teamMailboxRole: TeamMemberRole): SMono[Unit] =
     SMono(mailboxManager.applyRightsCommandReactive(path,
       MailboxACL.command
         .forUser(user)
-        .rights(TEAM_MAILBOX_RIGHTS_DEFAULT)
+        .rights(rightsToAdd(teamMailboxRole))
         .asAddition(),
       session))
+      .`then`(removeAdministerRightIfNeeded(path, user, session, teamMailboxRole))
       .`then`()
```

Comment:
I think we have a mailbox ACL command for resetting rights (without add/remove).

CF 

```
private static Mono<Void> applyRight(MailboxManager mailboxManager, MailboxSession mailboxSession, String identifier, MailboxPath mailboxPath) {
        return Mono.from(mailboxManager.applyRightsCommandReactive(
            mailboxPath,
            MailboxACL.command().key(EntryKey.deserialize(identifier)).noRights().asReplacement(),
            mailboxSession));
    }
```

Reply:
I did not go for the `.asReplacement()` as it could remove other rights beside the rights set by TMail server.

e.g. BOB as a team mailbox manager with the `a` right, he can use `SETACL` to add the DeleteMailbox right for example for ALICE.

Then adding/promoting ALICE to the team mailbox using rights replacement would remove the above right.

Is it valid concern or overkill?

Reply:
Likely it is overkill as only the team mailbox owner (aka TMail server) can set the ACL:
`Setting ACL is only permitted to the owner of the mailbox`.

Reply:
> Is it valid concern or overkill?

IMO overkill

Also I prefer that we set the rights for TeamMailbox to the exact value, the conceptual model is easier.